test/image/mocks/bar_attrs_relative.json

```diff
@@ -2,27 +2,37 @@
     "data":[
         {
             "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3333333333,4],
+            "textposition":"outside",
             "y":[1,2,3,4],
             "x":[1,2,3,4],
             "type":"bar"
         }, {
             "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"outside text",0],
```

Comment:
> The way I understood the criteria for auto is that if it doesn't fit inside, it goes outside (no matter if shrinking is required to draw it outside).

Yeah, that's probably what I said :) But would you agree that if the shrinking is the same either way then it should stay inside?

> 0-size bars aren't even included in gd.calcdata

OK cool - yes, definitely an edge case, lets leave this one as it is.